### PR TITLE
Force clear the possible side effect after exporting an action

### LINKED ISSUE #172: Model gets heavily distorted after export
**OS:**
Manjaro Linux 18.0.3

**Godot version:**
`master` / 8b231b96

**Blender version:**
Blender : 2.8 beta / 84820e
Exporter : `blender2.8` / a14b44d

**Issue description:**
When I export a human model, it gets severely distorted after the process completes. The exported data seems to work correctly, but the Blender file needs to be reverted in order to continue working with it.

The model has number of blendshapes, but it's not clear if the fact has affected the symptom.

The original model:
image

And after the export:
image

### LINKED ISSUE #199: Animations influence others later in the NLA stack
**OS:**

linux

**Godot version:**

3.1-beta9

**Blender version:**

Blender 2.79 (sub 6)
	build hash: 2c0c1f494dcf

**Issue description:**

1. Set up several animations, each having keyframes for a different subset of bones
2. Push all animations onto the NLA stack
3. Notice that animations earlier in the stack influence animations later in the call stack

For example, here is my lean left animation:

lean

Here is my spin left animation:

spin_left

Here's what it looks like in Godot:

godot

**Minimal reproduction project:**

[example.zip](https://github.com/godotengine/godot-blender-exporter/files/3026559/example.zip)

### PR DESCRIPTION
**Pull request by Jason0214**
fixes #199 

if apply this patch onto blender2.8, hopeful it will also fix #172

### CODE REVIEW

File: io_scene_godot/converters/animation/animation_data.py [hunk 1]
```diff
@@ -175,6 +180,17 @@ def export_stashed_track(self, escn_file, export_settings, stashed_track):
...
                         )
+                        self.clear_action_effect()
+
+    def clear_action_effect(self):
+        """Clear side effect of exporting an action"""
+        if (isinstance(self.blender_object, bpy.types.Object) and
+                self.blender_object.pose is not None):
+            bpy.context.scene.objects.active = self.blender_object
+            bpy.ops.object.mode_set(mode='POSE', toggle=False)
+            bpy.ops.pose.select_all(action='SELECT')
```
Comment (rcorre) [inline io_scene_godot/converters/animation/animation_data.py]: This only works if all bones are unhidden. Try hiding the upper body bones in the example from #199, and you'll get the same result as before the patch.

This is still better than nothing though.

Comment (rcorre) [reply io_scene_godot/converters/animation/animation_data.py]: Something like https://blender.stackexchange.com/a/79507/60829 might work. I gave this a try and it seems to work without requiring bones to be unhidden:

```
    def clear_action_effect(self):
        """Clear side effect of exporting an action"""
        if (isinstance(self.blender_object, bpy.types.Object) and
                self.blender_object.pose is not None):

            for bone in self.blender_object.pose.bones:
                bone.rotation_quaternion = mathutils.Quaternion((0, 0, 0), 0)
                bone.scale = mathutils.Vector((1, 1, 1))
                bone.location = bone.bone.head
```

Comment (rcorre) [reply io_scene_godot/converters/animation/animation_data.py]: Scratch that, it completely messed up the exported animations :D

Comment (Jason0214) [reply io_scene_godot/converters/animation/animation_data.py]: Hmm..  maybe simply toggle every bone to unhidden in exporting and then toggle them back,
let me give a try

Comment (Jason0214) [reply io_scene_godot/converters/animation/animation_data.py]: well, I think your approach is right, just set the pose matrix to identity would work.

**Comment by rcorre** [comment]
The latest patch seems to work, thanks!

**Pull request merged**